connectors/sources/network_drive.py

```diff
@@ -255,8 +275,10 @@ def __init__(self, configuration):
         self.drive_path = self.configuration["drive_path"]
         self.drive_type = self.configuration["drive_type"]
         self.identity_mappings = self.configuration["identity_mappings"]
-        self.session = None
         self.security_info = SecurityInfo(self.username, self.password, self.server_ip)
+        self.smb_connection = SMBSession(
+            self.server_ip, self.username, self.password, self.port
+        )
```

Comment:
Can it raise an exception when it's created? If so, we cannot have it in the initialiser

Reply:
It should not raise error, but for safer side i'll update this and move it to a separate method and make it a cached_property like below, WDYT ?

```
    @cached_property
    def smb_connection(self):
        return SMBSession(
            self.server_ip, self.username, self.password, self.port
        )
```

Reply:
Yeah if it works, then it's a good idea. Does SMBSession attempt to access 3rd-party? Cause if so, we might need to make it async too, and then cached property will not work any more :/